function-aws/src/main/java/io/micronaut/function/aws/MicronautRequestHandler.java

```diff
@@ -125,7 +126,14 @@ public final O handleRequest(I input, Context context) {
         if (!inputType.isInstance(input)) {
             input = convertInput(input);
         }
-        return this.execute(input);
+        try {
+            O output = this.execute(input);
+            applicationContext.getEventPublisher(AfterExecutionEvent.class).publishEvent(AfterExecutionEvent.success(output));
```

Comment:
Need to cache the event publisher lookup since it can be expensive in terms of adding to the execution time

Reply:
actually the lookup is always happening only once either on success or on exception (if I skip the unlikely option that the lookup/publish fails)

Reply:
It will lookup the bean again when the lambda is reused.